File: app/src/main/java/com/bkjcb/rqapplication/datebase/ObjectBox.java
package com.bkjcb.rqapplication.datebase;

import android.content.Context;
import android.util.Log;

import com.bkjcb.rqapplication.BuildConfig;
import com.bkjcb.rqapplication.model.Emergency;
import com.bkjcb.rqapplication.contactBook.model.Level;
import com.bkjcb.rqapplication.model.MyObjectBox;
import com.bkjcb.rqapplication.contactBook.model.Unit;
import com.bkjcb.rqapplication.contactBook.model.User;

import io.objectbox.Box;
import io.objectbox.BoxStore;
import io.objectbox.android.AndroidObjectBrowser;

/**
 * Created by DengShuai on 2019/2/27.
 * Description :
 */
public class ObjectBox {
    private static BoxStore boxStore;
    private static Box<Emergency> emergencyBox;
    private static Box<Unit> unitBox;
    private static Box<Level> levelBox;
    private static Box<User> userBox;

    public static void init(Context context) {
        boxStore = MyObjectBox.builder()
                .androidContext(context.getApplicationContext())
                .build();
        if (BuildConfig.DEBUG) {
            boolean started = new AndroidObjectBrowser(boxStore).start(context);
            Log.i("ObjectBrowser", "Started: " + started);
        }
    }

    public static BoxStore get() {
        return boxStore;
    }

    public static Box<Emergency> getEmergencyBox() {
        if (emergencyBox == null) {
            emergencyBox = boxStore.boxFor(Emergency.class);
        }
        return emergencyBox;
    }

    public static Box<Unit> getUnitBox() {
        if (unitBox == null) {
            unitBox = boxStore.boxFor(Unit.class);
        }
        return unitBox;
    }

    public static Box<Level> getLevelBox() {
        if (levelBox == null) {
            levelBox = boxStore.boxFor(Level.class);
        }
        return levelBox;
    }

    public static Box<User> getUserBox() {
        if (userBox == null) {
            userBox = boxStore.boxFor(User.class);
        }
        return userBox;
    }

}

File: app/src/main/java/com/bkjcb/rqapplication/base/datebase/ObjectBox.kt
package com.bkjcb.rqapplication.base.datebase

import android.content.Context
import android.util.Log
import com.bkjcb.rqapplication.BuildConfig
import com.bkjcb.rqapplication.MyObjectBox
import com.bkjcb.rqapplication.contact.model.Level
import com.bkjcb.rqapplication.contact.model.Unit
import com.bkjcb.rqapplication.contact.model.User
import com.bkjcb.rqapplication.emergency.model.Emergency
import io.objectbox.Box
import io.objectbox.BoxStore
import io.objectbox.android.AndroidObjectBrowser

/**
 * Created by DengShuai on 2019/2/27.
 * Description :
 */
object ObjectBox {
    lateinit var boxStore: BoxStore
        private set

    fun getEmergencyBox(): Box<Emergency> = boxStore.boxFor(Emergency::class.java)
    fun getUnitBox(): Box<Unit> = boxStore.boxFor(Unit::class.java)
    fun getLevelBox(): Box<Level> = boxStore.boxFor(Level::class.java)
    fun getUserBox(): Box<User> = boxStore.boxFor(User::class.java)
    fun init(context: Context) {
        boxStore = MyObjectBox.builder()
                .androidContext(context.applicationContext)
                .build()
        if (BuildConfig.DEBUG) {
            val started = AndroidObjectBrowser(boxStore).start(context)
            Log.i("ObjectBrowser", "Started: $started")
        }
    }


}